### ISSUE TITLE
Screenreader NVDA is not able to read text in terminal

### ISSUE THREAD

**Issue by Pit-Storm** [microsoft/terminal#7766]
# Environment

```none
Windows build number: Win32NT             10.0.18362.0 Microsoft Windows NT 10.0.18362.0
Windows Terminal version (if applicable): 1.3.2651.0

Any other software?
* Screen magnifier (Windows 10 Built in)
* NVDA Screenreader
```

# Steps to reproduce
* Start NVDA Screenreader with Mouseover-Setting enabled
* Start Windows-Terminal

# Expected behavior
* When hovering over a line of text, the Screenreader should speak the text.

# Actual behavior
* The screenreader isn't recognizing any text.

**Comment by carlos-zamora** [comment]
This sounds like https://github.com/microsoft/microsoft-ui-xaml/issues/2182.

We have been working with NVDA to make sure Windows Terminal is accessible though. You should still be able to access the contents of Terminal with the keyboard.

**Comment by codeofdusk** [comment]
@michaelDCurran NVDA supports mouse events in UIA controls, right?

**Comment by codeofdusk** [comment]
I'm not a mouse user, but I've observed that:

* In conhost (with UIA), all text in the console is read when I slide my finger on the trackpad. With UIA disabled NVDA reads text with line resolution when I slide my finger.
* In Word with UIA, text is read with line resolution.

**Comment by codeofdusk** [comment]
OK, upon some more digging:

* By default, NVDA reads text under the mouse with paragraph resolution.
* Conhost doesn't seem to support the paragraph UIA unit, but it does comply with the spec (by providing the entire document, which is the next-largest supported unit).

Question is, what should a "paragraph" be in the console?

**Comment by Pit-Storm** [comment]
Oh wow! This is rly quick and lot response.

In old powershell it reads line by line. But could be more intuitive if a paragraph is a complete input line (wd with typed command) and a new paragraph the complete related output?

**Comment by codeofdusk** [comment]
What about, say, screen-based editors? I'm worried about this.

**Comment by Pit-Storm** [comment]
Good point. Didn't thought about it like that.

Maybe a newline character will indicate a new paragraph. But I don't now if there will be a newline character only for line wrapping.

Maybe the best thing is to fall back to read line by line, when paragraph based reading under mouse pointer is enabled. A lot of apps doing it like that.
It is way not optimal, but a more better solution as of don't handing any text over to NVDA.

**Comment by MAGOO56** [comment]
Is there any update on this issue? The last comment was from over 6 months ago and I haven't switched to Terminal for this reason alone. Otherwise, I've found a solution for most other items. I figure I'm in a minority where I'm still use to using the mouse and now find I can no longer read without a screen reader. For now, I'll continue with the console host.

**Comment by codeofdusk** [comment]
Where is the issue? If you'd like NVDA to read mouse tracked lines at line resolution:

1. Go to NVDA menu, preferences, settings.
2. Select "mouse" from the categories list.
3. Set "text unit resolution" to "line".
4. Verify if line tracking works as expected in terminal.

**Comment by MAGOO56** [comment]
Yes, I’ve tried it and it doesn’t work. I’ve also tried setting it to word. I expect that the line (or word) I hovered over would be read. This works in console host. The best I’ve been able to do so far in Terminal is to highlight the text, but that is cumbersome for more than a word or two. I have the latest versions ove terminal from Microsoft Store and the current production release of NVDA. Randy McGee From: Bill DenglerSent: Saturday, May 29, 2021 15:27To: microsoft/terminalCc: Randy McGee; CommentSubject: Re: [microsoft/terminal] Screenreader NVDA is not able to read text in terminal (#7766) Where is the issue? If you'd like NVDA to read mouse tracked lines at line resolution:Go to NVDA menu, preferences, settings.Select "mouse" from the categories list.Set "text unit resolution" to "line".Verify if line tracking works as expected in terminal.—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by zadjii-msft** [comment]
Punting out of 1.14 to the backlog, since this is still blocked on the upstream issue.

**Comment by hwf1324** [comment]
Fixed by https://github.com/nvaccess/nvda/pull/20450
